Nimble: BD address handling

andrzej-kaczmarek opened the pull request:
- ble_addr_t type provided instead of type/address pair (for now used in all public APIs)
- clean up usage of address type and own address type symbols (they were mixed in different parts of code)
- fix pairing with privacy enabled (i.e. we use RPA)
- add identity changed event (i.e. identity address received after pairing)

ccollins476ad commented:
Hi Andrzej,

Thanks, it looks good.  However, I am getting some compiler errors when I run "newt test net/nimble/host".  Could you please take a look?

Thanks,
Chris

andrzej-kaczmarek commented:
Yes, absolutely. I forgot to check unit tests on final version...

andrzej-kaczmarek commented:
I updated test cases and also fixed few small issues I did not see when testing manually.

There's one problem though: test vectors for SM SC test cases with privacy have to be recalculated since they were using invalid address types. I just commented them out for now, will try to update them in upcoming days...

ccollins476ad commented:
It looks good.  Thanks for the late night, Andrzej!
